pennylane/tape/qscript.py

```diff
@@ -652,7 +652,7 @@ def get_parameters(
                 params.extend(m.obs.data)
         return params
 
-    def set_parameters(self, params, trainable_only=True):
+    def set_parameters(self, params, trainable_only=True, disable_check_batching=False):
```

Comment:
This change will be affected by https://github.com/PennyLaneAI/pennylane/pull/4220 where parameters are fed into the Operator instantiation. One probably has to find a way to disable batch checking there.